### PR TITLE
Union constructors

### PR DESCRIPTION
**Pull request by DerThorsten**

### CODE REVIEW

File: include/sparrow/layout/union_array.hpp [hunk 1]
```diff
@@ -327,6 +491,89 @@ namespace sparrow
...
+        ArrowArray arr = make_arrow_array(
+            static_cast<std::int64_t>(size), // length
+            static_cast<std::int64_t>(null_count),
+            0, // offset
+            std::move(arr_buffs),
+            static_cast<std::size_t>(n_children), // n_children
+            child_arrays, // children
+            nullptr // dictionary
+        );
+        return arrow_proxy{std::move(arr), std::move(schema)};
```
[the remark this answers is not in the document]
Comment (DerThorsten) [reply include/sparrow/layout/union_array.hpp:574]: there are subtle changes, for instance the null count is  computed differently, it has a different number of buffers.
Not sure if its worth the effort

File: include/sparrow/layout/union_array.hpp [hunk 2]
```diff
@@ -162,6 +207,39 @@ namespace sparrow
...
+                    ret += std::to_string(i) + ",";
+                }
+            }
+            else
+            {
+                for (const auto& v : range)
+                {
+                    ret += std::to_string(v) + ",";
+                }
+            }
```
[the remark this answers is not in the document]
Comment (DerThorsten) [reply include/sparrow/layout/union_array.hpp:232]: > ```c++
> auto&& view = range_size == 0 ? std::views::iota(0, n) : range;
> ```

this line is not working. both operands need to be of the same type

[the remark this answers is not in the document]
Comment (DerThorsten) [reply include/sparrow/layout/union_array.hpp]:
```C++
ret += std::accumulate(view | std::views::transform([](auto v) { return std::format("{},", v); }), std::string());
```
that part is also not working.
So I just use the format function but will keep to loops

File: include/sparrow/layout/union_array.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (DerThorsten) [reply include/sparrow/layout/union_array.hpp]: I extracted the function to build the child_to_type_id mapping, the rest I think it is not worth it

File: include/sparrow/layout/union_array.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (DerThorsten) [reply include/sparrow/layout/union_array.hpp:257]: ..removed format again because apple clang does not find it...

**Review by Alex-PLACET — APPROVED** [review]

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**